src/mappings/entities/authorization.ts

```diff
@@ -0,0 +1,78 @@
+import { Codec } from "@polkadot/types/types";
+import { Authorization } from "../../types";
+import {
+  getFirstKeyFromJson,
+  getFirstValueFromJson,
+  getTextValue,
+} from "../util";
+import { EventIdEnum, ModuleIdEnum } from "./common";
+
+enum AuthorizationEventType {
+  AuthorizationAdded = "AuthorizationAdded",
+  AuthorizationRevoked = "AuthorizationRevoked",
+  AuthorizationRejected = "AuthorizationRejected",
+  AuthorizationConsumed = "AuthorizationConsumed",
+}
+
+enum AuthorizationStatus {
+  Pending = "Pending",
+  Consumed = "Consumed",
+  Rejected = "Rejected",
+  Revoked = "Revoked",
+  Expired = "Expired",
+}
+
+const authorizationEventTypes = new Set<string>(
+  Object.values(AuthorizationEventType)
+);
+
+const isAuthorizationEvent = (e: string): e is AuthorizationEventType =>
+  authorizationEventTypes.has(e);
+
+const authorizationEventStatusMapping = new Map<
+  AuthorizationEventType,
+  AuthorizationStatus
+>([
+  [AuthorizationEventType.AuthorizationConsumed, AuthorizationStatus.Consumed],
+  [AuthorizationEventType.AuthorizationRevoked, AuthorizationStatus.Revoked],
+  [AuthorizationEventType.AuthorizationRejected, AuthorizationStatus.Rejected],
+]);
+
+export async function mapAuthorization(
+  block_id: number,
+  event_id: EventIdEnum,
+  module_id: ModuleIdEnum,
+  params: Codec[]
+): Promise<void> {
+  if (module_id === ModuleIdEnum.Identity && isAuthorizationEvent(event_id)) {
+    if (authorizationEventStatusMapping.has(event_id)) {
+      await Authorization.create({
+        id: params[2],
+        block_id,
+        auth_id: params[2],
+        to_key: getTextValue(params[1]),
+        status: authorizationEventStatusMapping.get(event_id),
+      }).save();
+    } else {
+      const expiry = getTextValue(params[5]);
+
+      let status = AuthorizationStatus.Expired;
+      if (!expiry || +expiry > new Date().getTime()) {
```

Comment:
As discussed in slack we can't process the expired status since this code runs once and never updates, therefore if someone queries the entity hours from now, it might be expired by then, the best we can do is allow people to filter out expired authentications.

This means that if a consumer depends on this status enum, they must now calculate if the status is expired on the client side

Reply:
Removed the code and status.

Reply:
Should we keep a granular level change-log (where we are either discontinuing something or changed the logic quite considerably) somewhere that can be useful for SDK/anybody consuming graphql directly later on ?

cc. @satyamakgec @raycar5 @polymath-eric

Reply:
I think we will have to write a `Migration guide` for consumers indeed

Reply:
I think we should have a doc where we can record these things as we go along, even if its in bullet point form, so we can go back later.